### DISCUSSION TITLE
Parameters for Degradation Mechanisms in LFP Battery

### DISCUSSION THREAD

**Discussion by Devanshi-2-003** [pybamm-team/PyBaMM#4126] [Q&A]
I have been working on the degradation mechanisms in LFP battery. Now as given in [this notebook](https://docs.pybamm.org/en/latest/source/examples/notebooks/models/coupled-degradation.html), the code is running with OKane parameter set for NCM chemistry. I wish to model it for LFP chemistry. I can extract the parameter values for negative electrode for approximation (because both have a graphite cathode) but cannot extract the values for positive electrode parameters. What should be the range of values I can take for these parameters:

param = pybamm.ParameterValues("Prada2013")
params = pybamm.ParameterValues("OKane2022")
param.update({"Negative electrode cracking rate":params["Negative electrode cracking rate"]}, check_already_exists=False)
param.update({"Negative electrode initial crack length [m]": 2e-08}, check_already_exists=False)
param.update({"Negative electrode partial molar volume [m3.mol-1]": 3.1e-06}, check_already_exists=False)
param.update({"Negative electrode Young's modulus [Pa]": 15000000000.0}, check_already_exists=False)
param.update({"Negative electrode Poisson's ratio": 0.3}, check_already_exists=False)
param.update({"Negative electrode Paris' law constant b": 1.12}, check_already_exists=False)
param.update({"Negative electrode Paris' law constant m": 2.2}, check_already_exists=False)
param.update({'Negative electrode LAM constant proportional term [s-1]': 2.7778e-07}, check_already_exists=False)
param.update({'Negative electrode critical stress [Pa]': 60000000.0}, check_already_exists=False)
param.update({'Negative electrode LAM constant proportional term': 2.7778e-07 }, check_already_exists=False)
param.update({'Negative electrode LAM constant exponential term': 2.0 }, check_already_exists=False)
param.update({'Positive electrode LAM constant proportional term [s-1]': 2.7778e-07 }, check_already_exists=False)
param.update({'Positive electrode LAM constant exponential term': 2.0 }, check_already_exists=False)
param.update({"Positive electrode partial molar volume [m3.mol-1]": 1.25e-05}, check_already_exists=False)
param.update({"Positive electrode Young's modulus [Pa]": 375000000000.0}, check_already_exists=False)
param.update({"Positive electrode Poisson's ratio": 0.2}, check_already_exists=False)
param.update({"Positive electrode Paris' law constant b": 1.12}, check_already_exists=False)
param.update({"Positive electrode Paris' law constant m": 2.2}, check_already_exists=False)
param.update({'Positive electrode critical stress [Pa]': 375000000.0}, check_already_exists=False)
param.update({'Negative electrode reference concentration for free of deformation [mol.m-3]': 0.0}, check_already_exists=False)
param.update({'Positive electrode reference concentration for free of deformation [mol.m-3]': 0.0}, check_already_exists=False)
param.update({'Initial inner SEI thickness [m]': 0.0}, check_already_exists=False)
param.update({'Initial outer SEI thickness [m]': 5e-09}, check_already_exists=False)
param.update({'Typical plated lithium concentration [mol.m-3]': 1000.0}, check_already_exists=False)
param.update({'Lithium metal partial molar volume [m3.mol-1]': 1.3e-05}, check_already_exists=False)
param.update({'Negative electrode number of cracks per unit area [m-2]': 3180000000000000.0}, check_already_exists=False)
param.update({'Positive electrode number of cracks per unit area [m-2]': 3180000000000000.0}, check_already_exists=False)
param.update({'Negative electrode initial crack width [m]': 1.5e-08}, check_already_exists=False)
param.update({'Positive electrode initial crack width [m]': 1.5e-08}, check_already_exists=False)
param.update({'SEI resistivity [Ohm.m]': 200000.0}, check_already_exists=False)
param.update({'Inner SEI reaction proportion': 0.0}, check_already_exists=False)
param.update({'SEI growth activation energy [J.mol-1]': 38000.0}, check_already_exists=False)
param.update({'Outer SEI solvent diffusivity [m2.s-1]': 2.5000000000000002e-22}, check_already_exists=False)
param.update({'Bulk solvent concentration [mol.m-3]': 2636.0}, check_already_exists=False)
param.update({'Exchange-current density for stripping [A.m-2]': params["Exchange-current density for stripping [A.m-2]"]}, check_already_exists=False)
param.update({'Exchange-current density for plating [A.m-2]': params["Exchange-current density for plating [A.m-2]"]}, check_already_exists=False)
param.update({'Inner SEI partial molar volume [m3.mol-1]': 9.585e-05}, check_already_exists=False)
param.update({'Ratio of lithium moles to SEI moles': 1}, check_already_exists=False)
param.update({'Outer SEI partial molar volume [m3.mol-1]': 9.585e-05}, check_already_exists=False)
param.update({'Dead lithium decay rate [s-1]': 1e-06}, check_already_exists=False)
param.update({'Initial plated lithium concentration [mol.m-3]': 0.0}, check_already_exists=False)
param.update({'Negative electrode volume change': params["Negative electrode volume change"]}, check_already_exists=False)
 param.update({'Cell thermal expansion coefficient [m.K-1]': 1.1e-06}, check_already_exists=False)
param.update({'Lithium plating transfer coefficient': 0.65}, check_already_exists=False)
param.update({'Lithium plating kinetic rate constant [m.s-1]': 1e-09}, check_already_exists=False)

**Comment by BatteryFabricationCourse** [comment]
Hi, did you manage to figure out a way around this?

**Reply by Devanshi-2-003** [reply]
No, not yet. The values given here are for OKane set i.e. NMC chemistry. I wish to get the equations where these parameters are being used in calculating something so that I can have an idea how each parameter is contributing to the variables

**Reply by brosaplanella** [reply]
Unfortunately for LFP we have not been able to find a good parameter set in the literature. If you are aware of any please do let us know, and we can incorporate it to PyBaMM.

**Reply by nuanyuanyuanyuan** [reply]
@brosaplanella Hi, would you please tell me where to find all the exact values of these parameters as i do not get them from the documents  cited  in Prada2013 or OKane2022.

**Comment by Theoan0930** [comment]
Have you figure out this question? Recently I am also struggle on how to model LFP batttery.

**Comment by harisankarsuresh** [comment]
Hi I found through manual trial and error that using `"SEI": "interstitial-diffusion limited"` option for SEI u can change `'SEI lithium interstitial diffusivity [m2.s-1]'` parameter to change rate of degradation with respect to c-rate and `'SEI growth activation energy [J.mol-1]'` to change rate of degradation with respect to temperature. Try manually changing this to fit an experimental cycle aging data at different temperatures and C rates after optimising the discharge curves. You may arrive at an accurate degradation model.

**Comment by wandede0629** [comment]
Hello, I would like to research projects related to battery cycle acceleration testing. Do you have relevant experience in this area?
